internals/docs/guides/inspections.md

```diff
@@ -1,3 +1,114 @@
+
+## Prerequisites
+
+Follow [get started](https://docs.tizen.org/application/dotnet/get-started/overview/) page to setup Visual Studio for .NET development.
+
+To enable your application to use `Tizen.Inspections`:
+
+1. To use `Tizen.Inspections` to collect inspectable data, the application must be signed with a platform level distributor certificate. Refer this [guide](https://docs.tizen.org/application/dotnet/tutorials/certificates/getting-the-certificates/) on how to sign an application with the custom certificate.
+
+2. To use the methods and properties of the `Tizen.Inspections` namespace, include the namespace in your application::
+
+    ```c#
+    Using Tizen.Inspections
+    ```
+
+## Receive inspection events
+
+`Tizen.Inspections` allows you to receive notification in case of abnormal situation occurs on the platform (ex. process crashes). To do so, you must subscribe to the `EventReceived` event:
+
+```c#
+void callback(object sender, EventReceivedEventArgs e)
+{
+    ...
+}
+
+void getInspectableDataFromEvents()
+{
+    Inspector.EventReceived += callback;
+}
+```
+
+When a new event arrives, callback is called. To read the inspection data out of the inspection context use `GetInspectableData()`:
+
+```c#
+void callback(object sender, EventReceivedEventArgs e)
+{
+    if (e.Context.ModuleID == "org.tizen.system.crash")
+    {
+        string[] parameters = {"cs_info_json"};
+        InspectionData data = e.Context.GetInspectableData(parameters);
+    }
+
+    ...
+}
+```
+
+Use the `parameters` argument to specify which data to read from the inspection context. Available parameters depend on the module that provides the context.
+
+In case of `"org.tizen.system.crash"` module, the proper values are:
+
+-   `"cs_info_json"`: JSON report is returned, which contains information about crashed process such as callstack, mapped memory regions, and so on.
+-   `"cs_full"`: Full crash report ZIP archive is collected.
+
+The obtained data is associated with the `data` Inspections handler. To read the data content, use `Read()` method:
+
+```c#
+void callback(object sender, EventReceivedEventArgs e)
+{
+    if (e.Context.ModuleID == "org.tizen.system.crash")
+    {
+        string[] parameters = {"cs_info_json"};
+        InspectionData data = e.Context.GetInspectableData(parameters);
+
+        byte[] buf = new byte[100];
+        while (true)
+        {
+            // Read data and check for EOF
+            if (data.Read(buf, 0, buf.Length) == 0)
+                break;
+
+            // Log received data
+            Log.Info("InspectionsExample", System.Text.Encoding.UTF8.GetString(buf));
+        }
+    }
+}
+```
+
+## Request inspection data
+
+`Tizen.Inspections` can also request data from other application or service, especially the inspection data that for various reasons cannot or must not be saved in the logs, such as binary or in custom format data, large volume data, reports that are time-consuming to create, and so on.
+
```

Comment:
Instead of "inspection data", "diagnostic data" may be better for showing this API's purpose.

I am not sure following description is correct.
 - "for various reasons cannot or must not be saved in the logs, such as binary or in custom format data, large volume data, reports that are time-consuming to create, and so on."

The purpose to provide this diagnostic API interface is to collect "diagnostics data".
In most case, this API will be used by "diagnostic aggregator (& sender)" or "diagnostic analyzer"
Especially, "diagnostic aggregator" can send the diagnostic data to cloud like ELK(Elasticsearch, Logstash, Kibana).
(In this case, cloud will analyze the diagnostic data)

Thus, any diagnostic data can be provided.

Reply:
Removed the inaccurate description.